src/services/__tests__/anchor-service.test.ts

```diff
@@ -802,5 +728,53 @@ describe('anchor service', () => {
         blockchainService.sendTransaction = original
       }
     })
+
+    test('fail a batch if can not import Merkle CAR to IPFS', async () => {
+      const numRequests = 4
+      const requests = await fake.multipleRequests(numRequests)
+
+      const batch = new MockQueueMessage({
+        bid: uuidv4(),
+        rids: requests.map(({ id }) => id),
+      })
+      anchorBatchQueueService.receiveMessage.mockReturnValue(Promise.resolve(batch))
+
+      const original = blockchainService.sendTransaction
+      blockchainService.sendTransaction = () => {
+        return Promise.resolve(fakeTransaction)
+      }
+
+      jest.spyOn(ipfsService, 'importCAR').mockImplementation(async () => {
+        throw new Error(`Can not import merkle CAR`)
+      })
+      await expect(anchorService.anchorRequests()).rejects.toThrow()
+      const retrieved = await requestRepository.findByIds(requests.map((r) => r.id))
+      expect(retrieved.every((r) => r.status === RequestStatus.PENDING)).toBeTruthy()
+      blockchainService.sendTransaction = original
```

Comment:
is a try finally block not needed?

Reply:
There is no error thrown here, so I don't think "finally" is needed.